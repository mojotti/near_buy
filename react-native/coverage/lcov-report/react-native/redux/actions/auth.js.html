<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/redux/actions/auth.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">react-native/redux/actions</a> auth.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19</td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const login = <span class="fstat-no" title="function not covered" >(u</span>sername, token) =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        type: 'LOGIN',
        username: username,
        token: token
    };
};
&nbsp;
export const logout = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    return {</span>
        type: 'LOGOUT'
    };
};
&nbsp;
export const signup = <span class="fstat-no" title="function not covered" >(u</span>sername, password) =&gt; {
<span class="cstat-no" title="statement not covered" >    return <span class="fstat-no" title="function not covered" >(d</span>ispatch) =&gt; {</span>
    };
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
