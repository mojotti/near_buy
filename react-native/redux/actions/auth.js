export const login = (username, hash) => {
    return {
        type: 'LOGIN',
        username: username,
        hash: hash
    };
};

export const logout = () => {
    return {
        type: 'LOGOUT'
    };
};

export const signup = (username, password) => {
    return (dispatch) => {
    };
};
