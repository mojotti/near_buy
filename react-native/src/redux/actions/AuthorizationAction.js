export const login = (username, token, id) => {
  return {
    type: 'LOGIN',
    username,
    token,
    id,
  };
};

export const logout = () => {
  return {
    type: 'LOGOUT',
  };
};
