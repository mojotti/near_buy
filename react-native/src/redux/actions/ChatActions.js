import { createNewChat, getChatsForUser } from '../../networking/ApiCalls';

export const createChatRequestAction = () => {
  return {
    type: 'CREATE_CHAT_REQUEST',
  };
};

export const createChatSuccessAction = () => {
  return {
    type: 'CREATE_CHAT_SUCCESSFUL',
  };
};

export const createChatErrorAction = (error) => {
  return {
    type: 'CREATE_CHAT_FAILURE',
    error,
  };
};

export const fetchChatsRequestAction = () => {
  return {
    type: 'FETCH_CHATS_REQUEST',
  };
};

export const fetchChatsSuccessAction = (chatHeaders) => {
  return {
    type: 'FETCH_CHATS_SUCCESS',
    chatHeaders,
  };
};

export const fetchChatsActionError = (error) => {
  return {
    type: 'FETCH_CHATS_ERROR',
    error,
  };
};

export const requestChatsAction = (token) => {
  return (dispatch) => {
    dispatch(fetchChatsRequestAction());
    handleChatFetching(token, dispatch);
  };
};

export const handleChatFetching = (token, dispatch) => {
  getChatsForUser(token)
    .then((response) => {
      if (response === 'error') {
        dispatch(fetchChatsActionError(response));
      } else if (response === 'could not get chats') {
        dispatch(fetchChatsActionError('could not get chats'));
      } else {
        dispatch(fetchChatsSuccessAction(response));
      }
    });
};

export const createChatAction = (itemDetails, token) => {
  return (dispatch) => {
    dispatch(createChatRequestAction());
    handleChatCreation(dispatch, itemDetails, token);
  };
};

export const handleChatCreation = (dispatch, itemDetails, token) => {
  createNewChat(itemDetails, token)
    .then((response) => {
      if (response === 'chat created') {
        dispatch(createChatSuccessAction());
        dispatch(requestChatsAction(token));
      } else {
        dispatch(createChatErrorAction(response));
      }
    })
    .catch((error) => dispatch(createChatErrorAction(error)));
};

