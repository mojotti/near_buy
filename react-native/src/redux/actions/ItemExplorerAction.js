export const fetchItemsRequestAction = () => {
  return {
    type: 'FETCH_ITEMS_REQUEST',
  };
};

export const fetchItemsSuccessAction = () => {
  return {
    type: 'FETCH_ITEMS_SUCCESS',
  };
};

export const fetchItemsErrorAction = () => {
  return {
    type: 'FETCH_ITEMS_FAILURE',
  };
};

