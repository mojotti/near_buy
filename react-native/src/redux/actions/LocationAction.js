export const updateLocation = (coords) => {
  return {
    type: 'UPDATE_LOCATION',
    coords,
  };
};
