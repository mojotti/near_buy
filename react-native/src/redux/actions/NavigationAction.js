export const navigateToItem = (itemName) =>Â {
  return {
    type: 'NAVIGATE_TO_ITEM',
    itemName,
  };
};

