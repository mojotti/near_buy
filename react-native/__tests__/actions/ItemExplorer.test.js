import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { requestItemsAction } from '../../src/redux/actions/ItemExplorerAction';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});


const sellerId = 0;
const itemId = 0;
const token = 'fake token';
const title = 'foo';

const ITEM = {
  sellerId,
  itemId,
  title,
};

describe('requestItemsAction', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('gets items', async () => {
    fetch.mockResponseSuccess({ items: [1, 2, 3] });

    await await store.dispatch(requestItemsAction(token));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('does not get items', async () => {
    fetch.mockResponseFailure('error');

    await await await store.dispatch(requestItemsAction(token));
    expect(store.getActions()).toMatchSnapshot();
  });
});
