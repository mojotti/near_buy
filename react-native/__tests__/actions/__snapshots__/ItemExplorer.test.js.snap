// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestItemsAction does not get items 1`] = `
Array [
  Object {
    "type": "FETCHING_ITEMS",
  },
  Object {
    "error": "error",
    "type": "FETCH_ITEMS_FAILURE",
  },
]
`;

exports[`requestItemsAction gets items 1`] = `
Array [
  Object {
    "type": "FETCHING_ITEMS",
  },
  Object {
    "items": Array [
      1,
      2,
      3,
    ],
    "type": "FETCH_ITEMS_SUCCESS",
  },
]
`;
